scripts/mbedtls_framework/build_tree.py

```diff
@@ -142,3 +142,16 @@ def is_mbedtls_3_6() -> bool:
         return False
     with open(os.path.join(root, 'include', 'mbedtls', 'build_info.h'), 'r') as f:
         return re.search(r"#define MBEDTLS_VERSION_NUMBER.*0x0306", f.read()) is not None
+
+def is_mbedtls_4_x() -> bool:
```

Comment:
Isn't this check not really future proof? I mean, I know that 4.x is the near future and we're going to deal with it for a while, but this check is going to fail in >= 5.x. At the same time we don't know yet if tests based on this check will change by that time...
Can't we instead live with only `is_mbedtls_3.6` and use `not is_mbedtls_3.6` in `collect_test_cases.py` in place of this new function?

Reply:
That's a good point. Thus let's stick with:
```
if build_tree.looks_like_mbedtls_root(project_root) and \
       not build_tree.is_mbedtls_3_6():
```
and no need to add `is_mbedtls_4_x`.

Reply:
Fixed and rebased, thanks.